code/game/gamemodes/objective.dm

```diff
@@ -1054,3 +1054,24 @@ GLOBAL_LIST_EMPTY(possible_items_special)
 	for(var/T in allowed_types)
 		var/datum/objective/X = T
 		GLOB.admin_objective_list[initial(X.name)] = T
+
+/datum/objective/contract
```

Comment:
I'm confused why you need this objective type if pretty much everything is handled by contract datum and you do not assign this objective at all.

Reply:
While not being a "traditional" objective, it still matches a lot of the criteria. We have a target, an objective owner (contract owner), and it makes use of the target find function. It also will be further extended when this'll likely be used for the contractor antag type in the future.